app/services/update_avatar_file_service.rb

```diff
@@ -0,0 +1,82 @@
+#frozen_string_literal: true
+
+require "digest/md5"
+require "open-uri"
+require "image_processing/mini_magick"
+
+STORAGE_PATH = Rails.root.join("storage", "validators", "avatars").to_s.freeze
```

Comment:
This path look a little messed up. There's a different path defined in config/storage, and on localhost the files are saved in root/storage, not in root/storage/validators/avatars.
There're some temporary files stored in storage/validators/avatars and storage/validators/avatars/tmp - before and after resizing, converting etc - but since both versions are temporary, why dividing to tmp and not tmp, or create so nested directories just for temporary files?